packages/jobs/src/core/Worker.ts

```diff
@@ -0,0 +1,188 @@
+// Used by the job runner to find the next job to run and invoke the Executor
+
+import console from 'node:console'
+import process from 'node:process'
+import { setTimeout } from 'node:timers'
+
+import type { BaseAdapter } from '../adapters/BaseAdapter'
+import type { BasicLogger } from '../types'
+
+import {
+  DEFAULT_MAX_ATTEMPTS,
+  DEFAULT_MAX_RUNTIME,
+  DEFAULT_SLEEP_DELAY,
+  DEFAULT_DELETE_FAILED_JOBS,
+} from './consts'
+import { AdapterRequiredError } from './errors'
+import { Executor } from './Executor'
+
+// The options set in api/src/lib/jobs.ts
+export interface WorkerConfig {
+  maxAttempts?: number
+  maxRuntime?: number
+  deleteFailedJobs?: boolean
+  sleepDelay?: number
+}
+
+// Additional options that the rw-jobs-worker process will set when
+// instantiatng the Worker class
+interface Options {
+  adapter: BaseAdapter
+  logger?: BasicLogger
+  clear?: boolean
+  processName?: string
+  queue?: string | null
+  forever?: boolean
+  workoff?: boolean
+}
+
+// The default options to be used if any of the above are not set
+interface DefaultOptions {
+  logger: BasicLogger
+  maxAttempts: number
+  maxRuntime: number
+  deleteFailedJobs: boolean
+  sleepDelay: number
+  clear: boolean
+  processName: string
+  queue: string | null
+  forever: boolean
+  workoff: boolean
+}
+
+export const DEFAULTS: DefaultOptions = {
+  logger: console,
+  processName: process.title,
+  queue: null,
+  clear: false,
+  maxAttempts: DEFAULT_MAX_ATTEMPTS,
+  maxRuntime: DEFAULT_MAX_RUNTIME, // 4 hours in seconds
```

Comment:
If someone ever changes `DEFAULT_MAX_RUNTIME` they'll 100% forget to update this comment 🙈 
For that reason some people prefer to name constants like this something like `DEFAULT_MAX_RUNTIME_4_HOURS`. If you don't like that (and I'm not sure \*I\* do) another option is to add a doc comment like `/** 4 hours in seconds */` to where `DEFAULT_MAX_RUNTIME` is defined. Then you can just hover the constant and you'll see its value. It's much more likely that people remember to update comments located directly above the value they're changing than comments in other places in the code (But of course unfortunately not guaranteed)

Reply:
The value used to be hardcoded in there before I cleaned stuff up and moved it to a constant, I just never removed the comment that went with it.

Reply:
^^ To my point about forgetting to update comments when the code changes 😁